napari/components/dims.py

```diff
@@ -204,6 +204,31 @@ def set_range(self, axis: int, _range: Sequence[Union[int, float]]):
             self.range = full_range
         self.last_used = axis
 
+    def _set_ranges(
+        self,
+        _ranges: Sequence[Sequence[Union[int, float]]],
+        axes: Sequence[int] = None,
```

Comment:
arguments are reverse compared to `set_range`

Reply:
I did that since `axes` is currently optional while `_ranges` is not, but can swap it back and make it non-optional if that is preferred.

I agree it is cleanest from an API standpoint to have just one public `dims.set_range`. If `axis` is an integer, we could use the existing `set_range` code path but use the `_set_ranges` one when `axis` is a sequence. Many NumPy functions have precedence for allowing both tuples or scalar for `axis` so having a "singular" name that allows a sequences of axes may not be too confusing to users.

Reply:
I would prefer sticking to single `dims.set_range` and handling the different execution inside the `set_range` implementation after checking the type and structure of `axis`. I've seen this in many places in NumPy source code. An example: https://github.com/numpy/numpy/blob/e94bbb87ba10ec2585f87af8e9a80afee2f89e04/numpy/fft/helper.py#L65